src/core/storage/storage.js

```diff
@@ -32,65 +73,119 @@ export default class GlobalStorage {
    * @param {string} key The storage key to set
    * @param {*} data The data to set
    */
-  set (key, data) {}
+  set (key, data) {
+    if (key === undefined || key === null || typeof key !== 'string') {
+      throw new AnswersStorageError('Invalid storage key provided', key, data);
+    }
+
+    if (typeof data === 'undefined') {
+      throw new AnswersStorageError('Invalid storage key provided', key, data);
```

Comment:
I believe this message should refer to storage. How about 'Data must not be undefined'